packages/flame/lib/src/components/core/component.dart

```diff
@@ -580,7 +580,7 @@ class Component {
     );
     child._parent = this;
     final game = findGame();
-    if (isMounted) {
+    if (isMounted && !child.isMounted) {
```

Comment:
I don't fully get this, if the `child` is mounted already we will run `children.add(child);` from the `else`, that doesn't sound correct?

Reply:
Ah yes,  the confusing part.  If you put a print statement in each section numerically (1 for isMount, 2 for else, and 3 for the bottom if), the order of load is 2,1,3,1,3. 

So if you use the flow I sent you in discord, it helps to follow, but: 

2 = add() is called in onload
1 = called again for onMount
3 = future returns which was call for 1
1 = reAddChildren call to reorder parent->child relationships (also where I think global priority could be added) 
3 = future returns which was call for 1 from reAddChildren
Done.

So in short, the _addChild gets called 5 times for every component to progress through the life cycle. 

That's what had me so confused all this time. That method is 1 incorrect line away from an infinite loop - ask me how I know, lol.

Reply:
So to clarify for myself when reading this again; this case is for when it comes in here from `detach`

Reply:
It's actually for any case in which the components have been loaded and mounted but now are not mounted. 

This error could come up where you detached a component but never tripped their remove events and you then add a new component (without ever leaving FlameGame) which would trip reAddChildren which then would throw the error.

Reply:
![Loading Diagram drawio](https://github.com/flame-engine/flame/assets/17042581/2372fbc5-0a80-4c35-890f-b03fa4e1c148)

Hopefully this helps others map out the loading process....

Reply:
So this is what clued me in:

Initial adding of the Rectangle:
```dart
flutter: _reAddChildren this: Instance of 'MyGame' and isLoaded: true and isMounted: true
flutter: _reAddChildren child: Instance of 'RectangleComponent' and isLoaded: true and isMounted: false
```
And this is when it was being reloaded:
```
flutter: _reAddChildren this: Instance of 'MyGame' and isLoaded: true and isMounted: true
flutter: _reAddChildren child: Instance of 'RectangleComponent' and isLoaded: true and isMounted: true
```

So originally it was passing an already mounted component to the queue for `updateTree` which only knows to mount, remove, or move parents for components.  As the rectangle was already mounted, it threw the error.

Reply:
> I don't fully get this, if the `child` is mounted already we will run `children.add(child);` from the `else`, that doesn't sound correct?

You are correct BUT `children.add` in this instance is just the `ComponentSet` and because `children` gets wiped every `reAddChildren` cycle, it never grows. as evidenced by this dump:

```dart
flutter: Load Page clicked
flutter: _addChild this: Instance of 'MyGame' and isLoaded: false and isMounted: false
flutter: _addChild child: Instance of 'RectangleComponent' and isLoaded: false and isMounted: false
flutter: Adding to Children and the length before is: 0
flutter: Adding to Children and the length after is: 1

flutter: Home Page clicked

flutter: Load Page clicked
flutter: _addChild this: Instance of 'MyGame' and isLoaded: true and isMounted: true
flutter: _addChild child: Instance of 'RectangleComponent' and isLoaded: true and isMounted: true
flutter: Adding to Children and the length before is: 0
flutter: Adding to Children and the length after is: 1
```